products/views.py

```diff
@@ -1,3 +1,78 @@
-from django.shortcuts import render
+import json, math
+from django.db.models.aggregates import Aggregate
+from django.db.models.query_utils import Q
 
-# Create your views here.
+from django.http.response         import JsonResponse
+from django.views                 import View
+from django.db.models             import Count
+
+from products.models              import Product
+from orders.models                import OrderStatus
+from users.utils                  import login_decorator
+
+class ProductView(View):
+    @login_decorator
+    def get(self, request):
+        user                = request.user
+        product_type        = request.GET.get('type')
+        character           = request.GET.get('character')
+        search              = request.GET.get('search')
+        category            = request.GET.get('category')
+        page                = int(request.GET.get('page', 0))
+        page_size           = int(request.GET.get('pageSize', 0))
+        order_conditions    = []
+        annotate_conditions = {}
+        filter_conditions   = Q()
```

Comment:
```suggestion
        q   = Q()
```
Q()을 사용할 때 위와 같이 소문자 q에 Q()를 담는 것이 암묵적인 방법입니다. 당연히 어떤 변수에 넣어도 상관 없지만, 협업하는 입장에서 장고에서 암묵적으로 가져가는 룰은 지키는 것이 좋다고 생각합니다.

Reply:
수정 완료했습니다